script/tool/test/create_all_plugins_app_command_test.dart

```diff
@@ -103,6 +108,97 @@ void main() {
           baselinePubspec.environment?[dartSdkKey]);
     });
 
+    test('macOS deployment target is modified in Podfile', () async {
+      final RepositoryPackage plugin = createFakePlugin('plugina', packagesDir);
+      final File podfileFile =
+          plugin.directory.childDirectory('macos').childFile('Podfile');
+      podfileFile.createSync(recursive: true);
+      podfileFile.writeAsStringSync("""
+platform :osx, '10.11'
+# some other line
+""");
+
+      await runCapturingPrint(runner, <String>['all-plugins-app']);
+      final List<String> podfile = command.app
+          .platformDirectory(FlutterPlatform.macos)
+          .childFile('Podfile')
+          .readAsLinesSync();
+
+      expect(
+          podfile,
+          everyElement((String line) =>
+              !line.contains('platform :osx') || line.contains("'10.15'")));
+    },
+        // Podfile is only generated (and thus only edited) on macOS.
+        skip: !io.Platform.isMacOS);
```

Comment:
> I added the skip because modification of the Podfile itself (in the actual command) only runs on macOS.

Oh right, sorry.

> Thinking about it, I could add another test verifying that the (mocked) Podfile indeed isn't changed on non-macOS platforms.

That's not really useful though; if the Podfile were generated unconditionally by `flutter` we would ideally change it unconditionally, so it's not a behavior we specifically want.